File: examples/TcpProbe.fs
open System.Net.Sockets

// OpenClaw TCP port probe (F#) — checks gateway nodes
let probe (host: string) (port: int) (timeoutMs: int) : bool =
    use client = new TcpClient()
    try
        let task = client.ConnectAsync(host, port)
        task.Wait(timeoutMs) && client.Connected
    with _ -> false

[<EntryPoint>]
let main _ =
    let nodes = [ ("localhost", 8080); ("localhost", 8081) ]
    for (host, port) in nodes do
        let status = if probe host port 3000 then "OK  " else "FAIL"
        printfn "%s %s:%d" status host port
    0

File: examples/TcpProbe.cs
using System;
using System.Net.Sockets;

// OpenClaw TCP port probe (C#) — checks gateway nodes
class TcpProbe
{
    static bool Probe(string host, int port, int timeoutMs = 3000)
    {
        using var client = new TcpClient();
        try
        {
            var task = client.ConnectAsync(host, port);
            return task.Wait(timeoutMs) && client.Connected;
        }
        catch
        {
            return false;
        }
    }

    static void Main()
    {
        var nodes = new[] { ("localhost", 8080), ("localhost", 8081) };
        foreach (var (host, port) in nodes)
            Console.WriteLine($"{(Probe(host, port) ? "OK  " : "FAIL")} {host}:{port}");
    }
}
